src/inmanta/resources.py

```diff
@@ -310,6 +310,8 @@ def map_field(
                 return value
             except proxy.UnknownException as e:
                 return e.unknown
+            except KeyError as e:
```

Comment:
I find it a bit strange to catch any `KeyError` below this method. While thinking on this I came to wonder if our interpretation of this ticket is even correct? @edvgui from the example in the test case I get the impression that this is a bug in the resource implementation (it should raise a proper exception). The responsibility of core in my opinion is then to make sure to wrap whatever exception comes out of this with a compiler exception that states something like "Failed to get ...: <e>". But I must admit my experience here is limited, what do you think?

Reply:
Added this to https://github.com/inmanta/inmanta-core/issues/3388

Reply:
I don't mean to say that `KeyError` is not a proper exception, but I do think catching `KeyError` here and handling that case is way too specific. I think it would be better to catch any exception instead and wrap it in a `ExternalException` (subclass of `CompilerException`) to get an output along the lines of `Failed to get attribute ... on ... Caused by: KeyError: "ji"` with some location information in there. Similar to how the `ExplicitPluginException` and `WrappingRuntimeException` work for plugins.
I'm not sure at the moment how much work that would be (it's part of what #3388 is all about). An intermediate solution could be to just find out how core loses the exception information and reports only the message and fix that. I think the proper wrapping should be feasible but I've been known to be too optimistic about these things. This could be something to discuss on the core planning meeting this afternoon but that doesn't really concern you. What I mainly want to know from you is if you agree that what you really want is that when any exception (not just a `KeyError`) gets raised at any point below this method (not just for this specific dict access), that the exception gets reported properly.

Reply:
> What I mainly want to know from you is if you agree that what you really want is that when any exception (not just a KeyError) gets raised at any point below this method (not just for this specific dict access), that the exception gets reported properly.  

Yes I agree, and everything else you said above sounds good to me as well